### PR TITLE
fix a couple of actions and bump

### PR DESCRIPTION
**Pull request by shamilovtim**
- I noticed the `node:test` action failed on main from #436 where there was a bump to Node v20. 

- Also noticed that a couple of the CI steps were still using node v18, so I have updated those to also use the same setup as what was introduced in #436. 

- I also bumped the actions versions to newer ones

- There is a perceptible regression in node performance between node v18 and node v20 which caused the  `SyncManagerLevel` test timeout to be exceeded. By bumping the timeout of the `agent` test suite I was able to account for the larger test time. 

- I was able to parallelize `test-with-node` which reduced that CI step's runtime by about 8%

- The mocha junit reporter is obscuring test results on CI so failures are unknown at the moment. I think this is an outstanding problem, and the junit reporter is doing more harm than good.

### CODE REVIEW

File: .github/workflows/tests-ci.yml [hunk 1]
```diff
@@ -65,7 +56,7 @@ jobs:
           echo "DWN_SERVER_BACKGROUND_PROCESS=$!" >> $GITHUB_ENV
 
       - name: Run tests for all packages
-        run: pnpm --recursive --no-stream --sequential test:node -- --color --reporter mocha-junit-reporter --reporter-options mochaFile=./results.xml
+        run: npm run test:node --ws -- --color
```
Comment (shamilovtim) [inline .github/workflows/tests-ci.yml:59]: debugging this. something broke here with the upgrade to node (or hermit).

File: packages/agent/.mocharc.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
   "enable-source-maps": true,
   "exit": true,
-  "spec": ["tests/compiled/**/*.spec.js"]
+  "spec": ["tests/compiled/**/*.spec.js"],
+  "timeout": 5000
```
Comment (frankhinek) [inline packages/agent/.mocharc.json:5]:
```suggestion
  "spec": ["tests/compiled/**/*.spec.js"]
```
```suggestion
  "spec": ["tests/compiled/**/*.spec.js"],
  "timeout": 5000
```

File: .github/workflows/tests-ci.yml [hunk 3]
```diff
@@ -64,8 +55,11 @@ jobs:
           node node_modules/@web5/dwn-server/dist/esm/src/main.js &
           echo "DWN_SERVER_BACKGROUND_PROCESS=$!" >> $GITHUB_ENV
 
+      - name: Build tests for all packages
+        run: pnpm --recursive --stream --sequential build:tests:node
+
       - name: Run tests for all packages
-        run: pnpm --recursive --no-stream --sequential test:node -- --color --reporter mocha-junit-reporter --reporter-options mochaFile=./results.xml
+        run: pnpm --recursive --stream exec c8 mocha -- --color --reporter mocha-junit-reporter --reporter-options mochaFile=./results.xml
```
Comment (frankhinek) [inline .github/workflows/tests-ci.yml:62]:
```suggestion
        run: npm run test:node --ws -- --color
```

File: .github/workflows/tests-ci.yml [hunk 4]
```diff
@@ -64,8 +55,11 @@ jobs:
           node node_modules/@web5/dwn-server/dist/esm/src/main.js &
           echo "DWN_SERVER_BACKGROUND_PROCESS=$!" >> $GITHUB_ENV
 
+      - name: Build tests for all packages
+        run: pnpm --recursive --stream --sequential build:tests:node
+
       - name: Run tests for all packages
-        run: pnpm --recursive --no-stream --sequential test:node -- --color --reporter mocha-junit-reporter --reporter-options mochaFile=./results.xml
+        run: npm run test:node --ws -- --color
```
Comment (frankhinek) [inline .github/workflows/tests-ci.yml:62]: Revert to Tim's changes now that timeouts are resolved
```suggestion
        run: pnpm --recursive --stream exec c8 mocha -- --color --reporter mocha-junit-reporter --reporter-options mochaFile=./results.xml
```

**Review by frankhinek — APPROVED** [review]

**Review by csuwildcat — APPROVED** [review]

**Pull request merged**